python/ray/tests/conftest_docker.py

```diff
@@ -141,3 +144,76 @@ def print_logs(self):
 @pytest.fixture
 def docker_cluster(head_node, worker_node):
     yield (head_node, worker_node)
+
+
+def run_in_container(cmd: List[str], container_id: str):
+    docker_cmd = ["docker", "exec", container_id] + cmd
+    print(f"Executing command: {docker_cmd}", time.time())
+    resp = subprocess.check_output(docker_cmd, stderr=subprocess.STDOUT, timeout=1000)
+    output = resp.decode("utf-8").strip()
+    print(f"Output: {output}")
+    return output
+
+
+IMAGE_NAME = "rayproject/ray:runtime_env_container"
+NESTED_IMAGE_NAME = "rayproject/ray:runtime_env_container_nested"
+
+
+@pytest.fixture
+def podman_docker_cluster():
+    start_container_command = [
+        "docker",
+        "run",
+        "-d",
+        "--privileged",
+        "-v",
+        "/var/run/docker.sock:/var/run/docker.sock",
+        "-v",
+        "/var/lib/containers:/var/lib/containers",
+        IMAGE_NAME,
+        "tail",
+        "-f",
+        "/dev/null",
+    ]
+    container_id = subprocess.check_output(start_container_command).decode("utf-8")
+    container_id = container_id.strip()
+
+    # Get group id that owns the docker socket file. Add user `ray` to
+    # group to get necessary permissions for pulling an image from
+    # docker's local storage into podman
+    ls_output = run_in_container(["ls", "-l", "/var/run/docker.sock"], container_id)
```

Comment:
this is going to be always the same as the group id of `/var/run/docker.sock` that is mounted in the test env. you do not have to get it inside the container.

you can also run `stat -c %g /var/run/docker.sock` instead of using regex parsing on the ls output, which can be easy to make mistakes.

Reply:
I've switched to `stat -c %g /var/run/docker.sock`, thanks for the pointer!